src/test/java/io/jenkins/plugins/coverage/CoveragePublisherPipelineTest.java

```diff
@@ -0,0 +1,169 @@
+package io.jenkins.plugins.coverage;
+
+import hudson.FilePath;
+import hudson.model.Descriptor;
+import io.jenkins.plugins.coverage.adapter.CoberturaReportAdapter;
+import io.jenkins.plugins.coverage.adapter.CoverageReportAdapter;
+import io.jenkins.plugins.coverage.adapter.JacocoReportAdapter;
+import io.jenkins.plugins.coverage.threshold.Threshold;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class CoveragePublisherPipelineTest {
+
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testAutoDetect() throws Exception {
+        CoverageScriptedPipelineScriptBuilder builder = CoverageScriptedPipelineScriptBuilder.builder();
+
+        WorkflowJob project = j.createProject(WorkflowJob.class, "coverage-pipeline-test");
+        FilePath workspace = j.jenkins.getWorkspaceFor(project);
+
+        workspace.child("cobertura-coverage.xml").copyFrom(getClass().getResourceAsStream("cobertura-coverage.xml"));
+
+        project.setDefinition(new CpsFlowDefinition(builder.build(), true));
+        WorkflowRun r = project.scheduleBuild2(0).waitForStart();
+        Assert.assertNotNull(r);
+
+        j.assertBuildStatusSuccess(j.waitForCompletion(r));
+        j.assertLogContains("Auto Detect was ended: Found 1 report", r);
+    }
+
+    public void testAdapters() throws Exception {
+        CoverageScriptedPipelineScriptBuilder builder = CoverageScriptedPipelineScriptBuilder.builder().
+                addAdapter(new CoberturaReportAdapter("cobertura-coverage.xml"))
+                .addAdapter(new JacocoReportAdapter("jacoco.xml"));
+
+
+        WorkflowJob project = j.createProject(WorkflowJob.class, "coverage-pipeline-test");
+        FilePath workspace = j.jenkins.getWorkspaceFor(project);
+
+        workspace.child("cobertura-coverage.xml").copyFrom(getClass().getResourceAsStream("cobertura-coverage.xml"));
+        workspace.child("jacoco.xml").copyFrom(getClass().getResourceAsStream("jacoco.xml"));
+
+        project.setDefinition(new CpsFlowDefinition(builder.build(), true));
+        WorkflowRun r = project.scheduleBuild2(0).waitForStart();
+
+        Assert.assertNotNull(r);
+
+        j.assertBuildStatusSuccess(j.waitForCompletion(r));
+        j.assertLogContains("A total of 2 reports were found", r);
+
+    }
+
+    public static class CoverageScriptedPipelineScriptBuilder {
```

Comment:
Maybe it was an overkill. It could be preferable to just have scripts, which can be later used for documentation purposes